### PR TITLE
diamond blastx subworkflow

### PR DESCRIPTION
**Pull request by alxndrdiaz**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/sanger-tol/blobtoolkit/tree/master/.github/CONTRIBUTING.md)
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by alxndrdiaz** [comment]
Small test works: `nextflow run . -profile test,docker --outdir <OUTDIR>`, and files to run the full test were also added

**Review by priyanka-surana — CHANGES_REQUESTED** [review]
How does this result get merged into the `blobdir`?

**Comment by priyanka-surana** [comment]
When I run the `test` profile, I get the following warning.
```
WARN: Found unexpected parameters:
* --uniprot_blastx: /lustre/scratch123/tol/teams/tolit/users/ps22/pipelines/blobtoolkit/assets/test/mCerEla1.1.buscoregions.dmnd
- Ignore this warning: params.schema_ignore_params = "uniprot_blastx" 
```

I think modifying the schema and config should fix it.

**Comment by alxndrdiaz** [comment]
> When I run the `test` profile, I get the following warning.
> 
> ```
> WARN: Found unexpected parameters:
> * --uniprot_blastx: /lustre/scratch123/tol/teams/tolit/users/ps22/pipelines/blobtoolkit/assets/test/mCerEla1.1.buscoregions.dmnd
> - Ignore this warning: params.schema_ignore_params = "uniprot_blastx" 
> ```
> 
> I think modifying the schema and config should fix it.

Done, the warning should not appear anymore.

File: modules/local/blobtoolkit/chunk.nf [hunk 1]
```diff
@@ -0,0 +1,36 @@
+process BLOBTOOLKIT_CHUNK {
+    tag "$meta.id"
+    label 'process_single'
+
+    if (workflow.profile.tokenize(',').intersect(['conda', 'mamba']).size() >= 1) {
+        exit 1, "BLOBTOOLKIT_CHUNK_BUSCO module does not support Conda. Please use Docker / Singularity / Podman instead."
```
Comment (priyanka-surana) [inline modules/local/blobtoolkit/chunk.nf:6]:
```suggestion
        exit 1, "BLOBTOOLKIT_CHUNK module does not support Conda. Please use Docker / Singularity / Podman instead."
```

File: workflows/blobtoolkit.nf [hunk 2]
```diff
@@ -117,6 +119,14 @@ workflow BLOBTOOLKIT {
 
     BUSCO_DIAMOND ( ch_genome, ch_taxon_taxa, ch_busco_db, ch_uniprot, params.blastp_outext, params.blastp_cols )
     ch_versions = ch_versions.mix ( BUSCO_DIAMOND.out.versions )
+    
+    RUN_BLASTX ( 
+        ch_genome.join(BUSCO_DIAMOND.out.first_table).map { meta, fasta, first_table -> [ meta, fasta ] },
+        ch_genome.join(BUSCO_DIAMOND.out.first_table).map { meta, fasta, first_table -> [ meta, first_table ] },
```
Comment (priyanka-surana) [inline workflows/blobtoolkit.nf:125]: Why are you doing the join and then split? Why not use the original version?

**Review by priyanka-surana — CHANGES_REQUESTED** [review]

File: workflows/blobtoolkit.nf (hunk 2 above)
Comment (alxndrdiaz) [reply workflows/blobtoolkit.nf]: Those transformations (join+split) would be only useful if there were multiple input genomes that would require a taxa file and identifiers to match each genome with a taxon, but as this is not implemented as far as I understood, this should work as well:

``` 
    RUN_BLASTX ( 
        ch_genome,
        BUSCO_DIAMOND.out.first_table,
        ch_blastx,
        params.blastx_outext,
        params.blastx_cols
    )
```

**Comment by alxndrdiaz** [comment]
@priyanka-surana Just want to confirm if this can be merged, the requested change should be resolved in this commit: [8f3f719](https://github.com/sanger-tol/blobtoolkit/pull/74/commits/8f3f719d409d6a2a07e95bc5394b7644a99d9c74). If there are no other changes it can be merged.

**Comment by priyanka-surana** [comment]
Closing and opening a fresh PR. Give a couple days.

**Pull request closed without merging**